app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/service/TrackerBlockingVpnService.kt

```diff
@@ -545,6 +569,64 @@ class TrackerBlockingVpnService : VpnService(), CoroutineScope by MainScope() {
         )
     }
 
+    private suspend fun monitorVpnAlwaysOnState() = withContext(Executors.newSingleThreadExecutor().asCoroutineDispatcher()) {
```

Comment:
What you think about keeping the VpnServiceStateLogger class and keeping all state related updates there. We can have something like:

interface VpnServiceStateLogger {
     fun logEnabled()
     fun logDisabled()
     fun logEnabling()
     fun monitorVpnAlwaysOnState()
     fun getCurrentState():  VpnServiceState
}


I think TrackerBlockVpnService is starting to be big so it might be a good idea to not keep putting more logic here unless it is really necessary.

Reply:
Not sure what you mean, do you mean just extracting that logic to a separate class but it not being a VPN callbacks plugin?

I agree VPN service is already somewhat big. Not sure about extracting that for now. I'd rather approach if/when extracting logic from the service more holistically (?)

Reply:
Yeah just extracting the logic to a separate class that we call in TrackerBlockingVpnService instead.

What do you mean by holistically? 

This is just a NIT anyway as I wanted that the service is big and hard to understand already.

Reply:
What I meant is having a dedicated task and there's probably more things we should/need to extract. So yeah, I am basically agreeing with you but wanted to have dedicated time for it, if that make sense.